auth/key/key.go

```diff
@@ -2,9 +2,19 @@ package key
 
 import (
 	logger "github.com/ipfs/go-log"
+	peer "github.com/libp2p/go-libp2p-core/peer"
 )
 
 var log = logger.Logger("key")
 
 // ID is the key identifier
-type ID string
+type ID = peer.ID
+
+// DecodeID parses an ID string
+func DecodeID(s string) (ID, error) {
+	pid, err := peer.Decode(s)
```

Comment:
Quick note that this is not strictly a peerID elsewhere and we already use pseudo peerIDs in some places. The initial implementation deliberately left out peerIDs and hence us having our own keybook implementation among other reasons.

Reply:
Totally, I'm mainly just piggybacking off the peer.Decode method, and explicitly coercing it to another type so we can refactor under the hood later. I'll add a note